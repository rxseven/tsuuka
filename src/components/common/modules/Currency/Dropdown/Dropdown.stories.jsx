import React from 'react';
import { storiesOf } from '@storybook/react';

import Dropdown from './index';

const title = 'Components/Common/Modules/Currency/Dropdown';
const props = {
  data: { code: 'USD', name: 'United States Dollar' },
  query: 'THB'
};
const component = <Dropdown {...props} />;

storiesOf(title, module).add('default', () => component);
