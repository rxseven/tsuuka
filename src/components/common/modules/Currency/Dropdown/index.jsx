import match from 'autosuggest-highlight/match';
import parse from 'autosuggest-highlight/parse';
import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';
import styled from 'styled-components';

import { genKey } from 'helpers/string';

const Code = styled.span`
  text-align: left;
  width: 2rem;
`;

const Item = styled.span`
  align-items: baseline;
  display: flex;
  justify-content: space-between;
`;

const Name = styled.span`
  display: inline-block;
`;

const Text = styled.span`
  color: ${({ theme, ...props }) =>
    props['data-highlight'] ? theme.color.primary : theme.color.base};
`;

const propTypes = exact({
  data: PropTypes.shape({
    code: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired,
  query: PropTypes.string.isRequired
});

function Dropdown({ data, query }) {
  const matches = match(data.name, query);
  const parts = parse(data.name, matches);
  const name = (
    <Name>
      {parts.map(part => {
        return (
          <Text
            data-highlight={part.highlight}
            key={`${genKey()}-${part.text}`}
          >
            {part.text}
          </Text>
        );
      })}
    </Name>
  );

  return (
    <Item>
      {name} <Code>{data.code}</Code>
    </Item>
  );
}

Dropdown.propTypes = propTypes;

export default Dropdown;
