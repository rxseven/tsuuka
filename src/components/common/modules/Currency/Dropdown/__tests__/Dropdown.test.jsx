import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Environment from 'tests/environment';
import Dropdown from '../index';

describe('<Currency.Dropdown />', () => {
  // Arrange
  const props = {
    data: { code: 'USD', name: 'United States Dollar' },
    query: 'THB'
  };
  const component = (
    <Environment>
      <Dropdown {...props} />
    </Environment>
  );

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
