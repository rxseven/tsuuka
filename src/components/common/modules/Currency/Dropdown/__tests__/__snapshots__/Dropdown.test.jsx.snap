// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Currency.Dropdown /> Snapshot tests should render correctly 1`] = `
<span
  className="sc-bwzfXH bMFRZP"
>
  <span
    className="sc-htpNat dbWHLM"
  >
    <span
      className="sc-bxivhb dqunTp"
      data-highlight={false}
    >
      United States Dollar
    </span>
  </span>
   
  <span
    className="sc-bdVaJa dzpzzo"
  >
    USD
  </span>
</span>
`;
